x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/ConnectorSyncJobIndexService.java

```diff
@@ -0,0 +1,169 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application.connector.syncjob;
+
+import org.elasticsearch.ResourceNotFoundException;
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.DocWriteRequest;
+import org.elasticsearch.action.get.GetRequest;
+import org.elasticsearch.action.get.GetResponse;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.action.support.WriteRequest;
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.client.internal.OriginSettingClient;
+import org.elasticsearch.xcontent.ToXContent;
+import org.elasticsearch.xpack.application.connector.Connector;
+import org.elasticsearch.xpack.application.connector.ConnectorFiltering;
+import org.elasticsearch.xpack.application.connector.ConnectorIndexService;
+import org.elasticsearch.xpack.application.connector.ConnectorIngestPipeline;
+import org.elasticsearch.xpack.application.connector.ConnectorTemplateRegistry;
+import org.elasticsearch.xpack.application.connector.syncjob.action.PostConnectorSyncJobAction;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.elasticsearch.xcontent.XContentFactory.jsonBuilder;
+import static org.elasticsearch.xpack.core.ClientHelper.CONNECTORS_ORIGIN;
+
+/**
+ * A service that manages persistent {@link ConnectorSyncJob} configurations.
+ */
+public class ConnectorSyncJobIndexService {
+
+    private static final Long ZERO = 0L;
+
+    private final Client clientWithOrigin;
+
+    public static final String CONNECTOR_SYNC_JOB_INDEX_NAME = ConnectorTemplateRegistry.CONNECTOR_SYNC_JOBS_INDEX_NAME_PATTERN;
+
+    /**
+     * @param client A client for executing actions on the connectors sync jobs index.
+     */
+    public ConnectorSyncJobIndexService(Client client) {
+        this.clientWithOrigin = new OriginSettingClient(client, CONNECTORS_ORIGIN);
+    }
+
+    /**
+     * @param request   Request for creating a connector sync job.
+     * @param listener  Listener to respond to a successful response or an error.
+     */
+    public void createConnectorSyncJob(
+        PostConnectorSyncJobAction.Request request,
+        ActionListener<PostConnectorSyncJobAction.Response> listener
+    ) {
+        try {
+            getSyncJobConnectorInfo(request.getId(), new ActionListener<>() {
+                @Override
+                public void onResponse(Connector connector) {
+                    Instant now = Instant.now();
+                    ConnectorSyncJobType jobType = Objects.requireNonNullElse(request.getJobType(), ConnectorSyncJob.DEFAULT_JOB_TYPE);
+                    ConnectorSyncJobTriggerMethod triggerMethod = Objects.requireNonNullElse(
```

Comment:
IMHO it's a bit better to have them here. As we've two constructors inside the request we need to keep two places in sync instead of one in the index service.